app/src/play/java/de/danoeh/antennapod/dialog/CustomMRControllerDialog.java

```diff
@@ -124,38 +165,89 @@ public View onCreateMediaControlView(Bundle savedInstanceState) {
-                            scale = (float) originalWidth / intrWidth;
+        LinearLayout.LayoutParams artParams;
+        if (landscape) {
+            artView = new ImageView(getContext()) {
+                @Override
+                protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+                    int desiredWidth = widthMeasureSpec;
+                    int desiredMeasureMode = MeasureSpec.getMode(heightMeasureSpec) == MeasureSpec.EXACTLY ?
+                            MeasureSpec.EXACTLY : MeasureSpec.AT_MOST;
+                    if (MeasureSpec.getMode(widthMeasureSpec) != MeasureSpec.EXACTLY) {
+                        Drawable drawable = getDrawable();
+                        if (drawable != null) {
+                            int intrHeight = drawable.getIntrinsicHeight();
+                            int intrWidth = drawable.getIntrinsicWidth();
+                            int originalHeight = MeasureSpec.getSize(heightMeasureSpec);
+                            if (intrHeight < intrWidth) {
+                                desiredWidth = MeasureSpec.makeMeasureSpec(
+                                        originalHeight, desiredMeasureMode);
+                            } else {
+                                desiredWidth = MeasureSpec.makeMeasureSpec(
+                                        Math.round((float) originalHeight * intrWidth / intrHeight),
+                                        desiredMeasureMode);
+                            }
                         }
-                        desiredHeight = MeasureSpec.makeMeasureSpec((int) (intrHeight * scale + 0.5f), MeasureSpec.EXACTLY);
                     }
+                    super.onMeasure(desiredWidth, heightMeasureSpec);
                 }
+            };
+            artParams = new LinearLayout.LayoutParams(
+                    ViewGroup.LayoutParams.WRAP_CONTENT,
+                    ViewGroup.LayoutParams.MATCH_PARENT);
+            MarginLayoutParamsCompat.setMarginStart(artParams,
+                    getContext().getResources().getDimensionPixelSize(R.dimen.media_router_controller_playback_control_horizontal_spacing));
+        } else {
+            artView = new ImageView(getContext()) {
+                @Override
+                protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+                    int desiredHeight = heightMeasureSpec;
+                    if (MeasureSpec.getMode(heightMeasureSpec) != MeasureSpec.EXACTLY) {
+                        Drawable drawable = getDrawable();
+                        if (drawable != null) {
+                            int originalWidth = MeasureSpec.getSize(widthMeasureSpec);
+                            int intrHeight = drawable.getIntrinsicHeight();
+                            int intrWidth = drawable.getIntrinsicWidth();
+                            float scale;
+                            if (intrHeight*16 > intrWidth*9) {
```

Comment:
can you maybe document what you are trying to achieve, here? which one of the imageview scaletypes [1] do you re-implement?

--
[1] https://developer.android.com/reference/android/widget/ImageView.ScaleType.html